app/src/components/common/form/big_number_input/index.tsx

```diff
@@ -35,20 +15,50 @@ const Input = styled.input`
     outline: none;
   }
 `
+export enum BigNumberInputError {
+  max,
+  min,
+  noError,
+}
 
-export const BigNumberInput = (props: Props) => {
+export interface BigNumberInputReturn {
+  name: string
+  value: BigNumber
+}
+
+type OverrideProperties<T, R> = Omit<T, keyof R> & R
+
+interface PropsBigNumber extends InputHTMLAttributes<HTMLInputElement> {
+  decimals: number
+  valueFixedDecimals?: number
+}
+
+type Props = OverrideProperties<
+  PropsBigNumber,
+  {
+    max?: BigNumber
+    min?: BigNumber
+    onChange: (value: BigNumberInputReturn) => void
+    onError?: (e: BigNumberInputError) => void
```

Comment:
This is overkill. The min/max properties are there mainly for disabling certain numbers, and for the arrows to behave in a sensible way. But error handling should be the parent's component responsibility.

Reply:
Errors are handled outside BigNumberInput, but I think BigNumberInput should inform there are errors to handle (?) so I added some code to do just that.

But yeah, it currently disables user input when there's an out of bounds error, and that's why https://github.com/protofire/gnosis-conditional-exchange/pull/626#issuecomment-623193010 happens.

That behavior was already in place, as a general "out of bounds" error. If we remove that, this is what will happen, and we should fix all of those problems individually:

![May-04-2020 17-29-43](https://user-images.githubusercontent.com/4015436/81010631-f0548e80-8e2c-11ea-890c-7b2f36533ca2.gif)